local-cli/generator-ios/templates/app/AppDelegate.m

```diff
@@ -15,36 +15,61 @@ @implementation AppDelegate
-
-  /**
-   * Loading JavaScript code - uncomment the one you want.
-   *
-   * OPTION 1
-   * Load from development server. Start the server from the repository root:
-   *
-   * $ npm start
-   *
-   * To run on device, change `localhost` to the IP address of your computer
-   * (you can get this by typing `ifconfig` into the terminal and selecting the
-   * `inet` value under `en0:`) and make sure your computer and iOS device are
-   * on the same Wi-Fi network.
-   */
-
-  jsCodeLocation = [NSURL URLWithString:@"http://localhost:8081/index.ios.bundle?platform=ios&dev=true"];
-
-  /**
-   * OPTION 2
-   * Load from pre-bundled file on disk. To re-generate the static bundle
-   * from the root of your project directory, run
-   *
-   * $ react-native bundle --minify
-   *
-   * see http://facebook.github.io/react-native/docs/runningondevice.html
-   */
-
-//   jsCodeLocation = [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
-
+    NSURL *jsCodeLocation = nil;
+    
+#ifdef DEBUG
+    
+    /**
+     
+     STARTING THE SERVER:
+     
+     From the repository root on the command line, type `npm start` (or `npm run hot` if you're using
+     the babel-plugin-react-transform hot-loader) and wait for it to say the packager is ready before
+     launching with one of the following options. You may also skip this and a server will be started
+     automatically when you build.
+     
+     1. DEBUG ON SIMULATOR:
+     - Leave useBundle NO (below) and optionally start the server (above).
+     - Be sure Chrome is open, then run on your preferred simulator. (Set this next to XCode's run button.)
+     - Chrome will prompt you to install the React Native plugin and open developer tools.
+     - If using the babel-plugin-react-transform hot-loader and see a red-box error on boot-up, click 'Dimiss (ESC)'
+     
+     2. DEBUG ON DEVICE:
+     - Open Network Utility from Spotlight and change computersWifiIP (below) your computer's WIFI IP address.
+     - Leave useBundle NO (below) and optionally start the server (above) and run on your device.
+     - Note that this may not work on all networks. Be sure your phone is on the same WIFI network as your computer.
+     
+     3. RUN STATIC BUILD ON SIM OR PHONE:
+     - Before starting the server, run `react-native bundle --minify` from the repository root.
+     (If you're using the hot-loader you should instead run `npm run bundle`.)
+     - Set useBundle YES (below) then optionally start your server (above) and run on your target.
+     
+     4. CREATE ARCHIVE FOR DISTRIBUTION:
+     - Do the bundling step from RUN STATIC BUILD above first.
+     - You don't need to change useBundle, that is automated for archiving. (Helpful for doing CI builds.)
+     - Go to Product > Archive then walk through the steps to export the build for your provisioning setup.
+     */
+    
+    BOOL useBundle = NO;
+    
+    __unused NSString *computersWifiIP = @"10.0.2.46";
```

Comment:
Sure, I'll change it to hostWifiIP and set the stub address to 0.0.0.0 to avoid confusion there. Thanks for the suggestion!